daemon/nmon/main_cmd.go

```diff
@@ -189,14 +180,13 @@ func (t *Manager) onSetNodeMonitor(c *msgbus.SetNodeMonitor) {
 			t.state.GlobalExpect = *c.Value.GlobalExpect
 			t.state.GlobalExpectUpdatedAt = time.Now()
 		}
+		return nil
 	}
 
-	doState()
-	doLocalExpect()
-	doGlobalExpect()
-
-	// inform the publisher we're done sending errors
-	sendError(nil)
+	err := errors.Join(doState(), doLocalExpect(), doGlobalExpect())
+	if c.Err != nil {
```

Comment:
Send() already tests if err is nil.
Test can be remove here for better readability.